src/renderer/features/task-detail/stores/taskExecutionStore.ts

```diff
@@ -461,11 +438,100 @@ export const useTaskExecutionStore = create<TaskExecutionStore>()(
         const effectiveRepoPath = taskState.repoPath;
 
         if (!effectiveRepoPath) {
-          store.addLog(taskId, {
-            type: "error",
-            ts: Date.now(),
-            message: "No repository folder selected.",
+          // Prompt user to select directory or clone
+          const hasRepo = !!task.repository_config;
+          const repoConfig = task.repository_config;
+
+          const result = await window.electronAPI.showMessageBox({
+            type: "question",
+            title: "Select working directory",
+            message:
+              hasRepo && repoConfig
+                ? `Do you have ${repoConfig.organization}/${repoConfig.repository} locally?`
+                : "Select a working directory for this task",
+            detail: hasRepo
+              ? "If you have the repository locally, we'll use that. Otherwise, we can clone it for you."
+              : "Choose a directory where the task will run.",
+            buttons: hasRepo
+              ? ["I have it locally", "Clone for me", "Cancel"]
+              : ["Select directory", "Cancel"],
+            defaultId: 0,
+            cancelId: hasRepo ? 2 : 1,
           });
+
+          if (result.response === (hasRepo ? 2 : 1)) {
+            // User cancelled
+            return;
+          }
+
+          if (result.response === 0) {
+            // User has repo locally or wants to select directory
+            const selectedPath = await window.electronAPI.selectDirectory();
+
+            if (!selectedPath) {
+              // User cancelled directory selection
+              return;
+            }
+
+            // Set the repo path and revalidate
+            store.setRepoPath(taskId, selectedPath);
+            await store.revalidateRepo(taskId);
+
+            // Retry running the task with the new path (skip initialization)
+            return store.runTask(taskId, task, true);
+          }
+
+          if (result.response === 1 && hasRepo && repoConfig) {
+            // User wants to clone - trigger clone and retry
+            const { repositoryWorkspaceStore } = await import(
```

Comment:
can we avoid inline imports

Reply:
I guess we do not need the store if not on this code path, but I like to avoid. Its a nit - feel free to ignore